clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -1171,4 +1253,43 @@ private void release() {
         if (refcount.decrementAndGet() == 0)
             currentThread.set(NO_CURRENT_THREAD);
     }
+
+    private class PatternSubscriptionTask implements DelayedTask {
+        private final long interval;
+
+        public PatternSubscriptionTask(long interval) {
+            this.interval = interval;
+        }
+
+        public void run(final long now) {
+            final List<String> topicsToSubscribe = new ArrayList<>();
+
+            final RequestFuture<ClientResponse> requestFuture = fetcher.sendMetadataRequest();
+            if (requestFuture != null) {
+                requestFuture.addListener(new RequestFutureListener<ClientResponse>() {
+                    @Override
+                    public void onSuccess(ClientResponse value) {
+                        MetadataResponse response =
```

Comment:
@SinghAsDev I think that works! I wasn't sure if we needed to handle errors in the MetadataResponse itself, but it looks like the topics are only added to the Cluster if it had no error in the response.